opentracing-vertx-web/src/main/java/io/opentracing/contrib/vertx/ext/web/WebSpanDecorator.java

```diff
@@ -0,0 +1,93 @@
+package io.opentracing.contrib.vertx.ext.web;
+
+import io.vertx.core.Handler;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.opentracing.BaseSpan;
+import io.opentracing.tag.Tags;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+
+/**
+ * Decorate server span at different stages. Do not call blocking code inside decorators!
+ *
+ * @author Pavol Loffay
+ */
+public interface WebSpanDecorator {
+    /**
+     * Decorate span when span is started.
+     *
+     * @param request server request
+     * @param span server span
+     */
+    void onRequest(HttpServerRequest request, BaseSpan<?> span);
+
+    /**
+     * Decorate span when span is rerouted.
+     *
+     * @param request server request
+     * @param span server span
+     */
+    void onReroute(HttpServerRequest request, BaseSpan<?> span);
+
+    /**
+     * Decorate span when the response is known. This is effectively invoked in BodyEndHandler which is added to
+     * - {@link io.vertx.ext.web.RoutingContext#addBodyEndHandler(Handler)}
+     *
+     * @param request server request
+     * @param span server span
+     */
+    void onResponse(HttpServerRequest request, BaseSpan<?> span);
```

Comment:
How does that work for chunked response ? Does it wait for the very last chunk and response ending ?